ts/image-occlusion/shapes/index.ts

```diff
@@ -0,0 +1,9 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+export { Shape } from "./base";
+export { Ellipse } from "./ellipse";
+export { extractShapesFromRenderedClozes } from "./from-cloze";
+export { Polygon } from "./polygon";
+export { Rectangle } from "./rectangle";
+export { Text } from "./text";
```

Comment:
I can definitely see that. I would also agree that the convenience factor in internal use really isn't that high. 

However,  given that third-party code consuming these APIs will likely also want to have access to their corresponding types, I'm wondering how we should best handle this here.

E.g., with the current state I can just import the various `Shape` types from this barrel file without having to worry about the internal structure of the `shapes` package (annotating shape types with e.g. `typeof globalThis.anki.imageOcclusion.Shape` doesn't work without some major TS acrobatics which are likely a bad practice of their own).

Perhaps just bundling type exports in barrel files like these could be an option? Or splitting out type exports in a different way like d.ts files?

FWIW, I don't feel particularly strong about this from a stylistic standpoint, just want to make sure that we go with as defensive a solution as possible, so that using `anki` types is not too brittle long-term.

Reply:
If you're consuming the types then it's arguably public API, and I remove my objection.

I'd like to return to the question of the best way to expose these to add-on authors in the future - some previous discussion about it can be found at https://github.com/ankitects/anki/issues/1764.